datasets/alt/alt.py

```diff
@@ -0,0 +1,407 @@
+                        d["status"] = "draft"
+                    else:
+                        d["status"] = "reviewed"
+                    data[sntid] = d
+                fin.close()
+
+            keys = {
+                "word_alignment": "word-alignment/data_ja.en-ja",
+                "en_tokenized": "word-alignment/data_ja.en-tok",
+                "jp_tokenized": "word-alignment/data_ja.ja-tok",
+            }
+            for k in keys:
+                file_path = os.path.join(basepath, f"Japanese-ALT-20170330/{keys[k]}")
+                fin = open(file_path, encoding="utf-8")
+                for line in fin:
+                    line = line.strip()
+                    sp = line.split("\t")
+                    _, urlid, sntid = sp[0].split(".")
+                    if urlid not in allow_urls:
+                        continue
+
+                    data[sntid][k] = sp[1]
+                fin.close()
+
+        elif self.config.name == "alt-my":
+            data = {}
+            for fname in ["data"]:
+                file_path = os.path.join(basepath, f"my-alt-190530/{fname}")
+                fin = open(file_path, encoding="utf-8")
+                for line in fin:
+                    line = line.strip()
+                    sp = line.split("\t")
+                    _, urlid, sntid = sp[0].split(".")
+                    if urlid not in allow_urls:
+                        continue
+
+                    data[sntid] = {"value": sp[1]}
+                fin.close()
+
+        elif self.config.name == "alt-km":
+            data = {}
+            for fname in ["data_km.km-tag.nova", "data_km.km-tok.nova"]:
+                file_path = os.path.join(basepath, f"km-nova-181101/{fname}")
+                fin = open(file_path, encoding="utf-8")
+                for line in fin:
+                    line = line.strip()
+                    sp = line.split("\t")
+                    _, urlid, sntid = sp[0].split(".")
+                    if urlid not in allow_urls:
+                        continue
+
+                    k = "km_pos_tag" if fname == "data_km.km-tag.nova" else "km_tokenized"
+                    if sntid in data:
+                        data[sntid][k] = sp[1]
+                    else:
+                        data[sntid] = {
+                            "km_pos_tag": None,
+                            "km_tokenized": None,
+                        }
+                        data[sntid][k] = sp[1]
+                fin.close()
+
+        elif self.config.name == "alt-my-transliteration":
+            file_path = os.path.join(basepath, f"my-en-transliteration/data.txt")
+            # Need to set errors='ignore' because of the unknown error
+            # UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte
+            # It might due to some issues related to Myanmar alphabets
+            fin = open(file_path, encoding="utf-8", errors="ignore")
```

Comment:
have you tried other encodings then utf-8 ?

Reply:
It has Myanmar characters. I have no idea what other encodings I should try but it should support utf-8 as default.